File: ABC297/e.py
n, k = map(int, input().split())
prices = sorted(map(int, input().split()))

def gen(len: int, min_: int):
    if len == 0:
        yield ()
        return
    for i in range(min_, n):
        for rest in gen(len - 1, i):
            yield (i, *rest)

billings = set()

for i, _ in enumerate(iter(object, None)):
    for p in gen(i + 1, 0):
        billings.add(sum(prices[j] for j in p))
    if len(billings) >= k:
        break

billings = sorted(billings)
print(billings[k - 1])

File: ABC297/e.rs
use std::collections;

fn gen(len: usize, min: usize, max: usize) -> Vec<Vec<usize>> {
    if len == 0 {
        return vec![vec![]];
    }
    let mut res = vec![];
    for i in min..max {
        let it = gen(len - 1, i, max);
        for mut v in it.into_iter() {
            let mut r = vec![i];
            r.append(&mut v);
            res.push(r);
        }
    }
    res
}

fn main() {
    proconio::input! {
        n: usize, k: usize,
        prices: [u64; n]
    }
    let mut billings = collections::HashSet::new();
    for i in 1.. {
        let it = gen(i, 0, n);
        for v in it {
            let sum = v.iter().map(|&i| prices[i]).sum::<u64>();
            billings.insert(sum);
        }
        if billings.len() >= k {
            break;
        }
    }
    let mut billings = billings.into_iter().collect::<Vec<_>>();
    billings.sort();
    println!("{}", billings[k - 1]);
}
